lib/Zonemaster/Backend/RPCAPI.pm

```diff
@@ -40,33 +40,51 @@ sub new {
     my $self = {};
     bless( $self, $type );
 
-    my $config = Zonemaster::Backend::Config->load_config();
-
-    if ( $params && $params->{db} ) {
-        eval {
-            eval "require $params->{db}";
-            die "$@ \n" if $@;
-            $self->{db} = "$params->{db}"->new( { config => $config } );
-        };
-        if ($@) {
-            handle_exception('new', "Failed to initialize the [$params->{db}] database backend module: [$@]", '001');
-        }
+    if ( ! $params || ! $params->{config} ) {
+        handle_exception('new', "Missing 'config' parameter", '000');
     }
-    else {
-        eval {
-            my $backend_module = "Zonemaster::Backend::DB::" . $config->BackendDBType();
-            eval "require $backend_module";
-            die "$@ \n" if $@;
-            $self->{db} = $backend_module->new( { config => $config } );
-        };
-        if ($@) {
-            handle_exception('new', "Failed to initialize the [$params->{db}] database backend module: [$@]", '002');
-        }
+
+    $self->{config} = $params->{config};
+
+    my $dbtype;
+    if ( $params->{db} ) {
+        $dbtype = $params->{db};
+    } else {
+        $dbtype = $self->{config}->BackendDBType();
     }
+    $dbtype = $self->_check_db($dbtype);
```

Comment:
Couldn't the call to _check_db() be replaced with something like `$dbtype = $self->{config}->check_db( $params->{db} );` a few lines up? The call to handle_exception() made sense back when we reloaded the config file all the time, but this PR fixes that. After this PR the only time _check_db() would call handle_exception() is in unit tests, and I don't think handle_exception is very helpful in that context.

Also don't you think `dbtype` is a better name than `db`? Since we don't have proper type annotations in Perl I try to convey nuances like that in the names themselves.

Reply:
I see what you mean for the `_check_db` call. I'll remove it.

> Also don't you think dbtype is a better name than db?

Do you mean in the hash passed to the constructor as in `params->{dbtype}` ?